src/processes/rename_labels.js

```diff
@@ -0,0 +1,80 @@
+const { BaseProcess } = require('@openeo/js-processgraphs');
+const Errors = require('../errors');
+
+module.exports = class rename_labels extends BaseProcess {
+
+
+    process(oldLabels, newLabels){
+        return function renameBandsInner(image) {
+            return image.select(oldLabels).rename(newLabels);
+        };
+    }
+
+    async execute(node) {
+        var dc = node.getArgument("data");
+        var dimensionName = node.getArgument("dimension");
+        var target = node.getArgument("target");
+        var source = node.getArgument("source");
+        var dimension = dc.getDimension(dimensionName);
+
+        if (!dc.hasDimension(dimensionName)) {
+            throw new Errors.ProcessArgumentInvalid({
+                process: this.spec.id,
+                argument: 'dimension',
+                reason: 'Dimension "' + dimension + '" does not exist.'
+            });
+        }
+
+        // ToDO: only bands is currently supported
+        if (dimension.type !== "bands") {
+            throw new Errors.ProcessArgumentInvalid({
+                process: this.spec.id,
+                argument: 'dimension',
+                reason: 'Only dimension "bands" is currently supported.'
+            });
+        }
+
+        var oldLabels;  // array for storing the old label names given by the user
+        var allOldLabels;  // array for storing the old existing label names
+        if (source !== undefined) {
+            oldLabels = source;
+            allOldLabels = Array.from(dimension.values); // deep copy is important
```

Comment:
Array.from doesn't support deep copies. It does shallow copies. What do you need here? Shallow or deep? @bgoesswe @claxn